src/dapp/components/ui/HalveningBanner.tsx

```diff
@@ -1,24 +1,56 @@
+import "./Banner.css";
+
+import { useService } from "@xstate/react";
 import React, { useState } from "react";
 
-import alert from "../../images/ui/expression_alerted.png";
 import closeIcon from "../../images/ui/close.png";
-
-import "./Banner.css";
+import alert from "../../images/ui/expression_alerted.png";
+import { service } from "../../machine";
+import { numberWithBreaks } from "../../utils/number";
+import { getNextHalvingThreshold } from "../../utils/supply";
 
 export const Banner: React.FC = () => {
   const [show, setShow] = useState(true);
```

Comment:
Would be nice to only show the banner as we approach the final ~25% of the next threshold